### PR TITLE
array

### PR DESCRIPTION
**Pull request by vialikanau-aliaksei**

### CODE REVIEW

File: array/js/array.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
+function isArray(object) {
+  return Array.isArray(object);
```
Comment (NikitaSedyx) [inline array/js/array.js]: it's cool you know Array method and can use it, but I want to see your implementation

Comment (NikitaSedyx) [reply array/js/array.js]: have seen below, please remove this function, it doesn't make sense

File: array/js/array.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+function isArray2(object) {
+  return object.constructor.name == "Array";
+}
+
+function isArray3(object) {
+  return {}.toString.call(object) == "[object Array]";
+}
+
+function range(start, end, step) {
+  if (end == undefined) {
```
Comment (NikitaSedyx) [inline array/js/array.js]: it's better to check ===
null == undefined = true

File: array/js/array.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+function isArray2(object) {
+  return object.constructor.name == "Array";
+}
+
+function isArray3(object) {
+  return {}.toString.call(object) == "[object Array]";
+}
+
+function range(start, end, step) {
```
Comment (NikitaSedyx) [inline array/js/array.js]: please look at requirements for function range(5) should return [0, 1, 2, 3, 4]
(1, 5) ->  [1, 2, 3, 4]

File: array/js/array.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  var array = [];
+  for (var i = start; i < end + 1; i = i + step) {
+    array.push(i);
+  }
+  return array;
+}
+
+function compact(array) {
+  return array.filter(function (element) {
+    return element != false;
```
Comment (NikitaSedyx) [inline array/js/array.js]: could you read what values are falsy? and how to use them in conditions

File: array/js/array.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+function sum2(array) {
+  var sum = 0;
+  for (var i = 0; i < array.length; i++) {
+    sum += array[i];
+  }
+  return sum;
+}
+
+function unique(array) {
+  var set = {};
```
Comment (NikitaSedyx) [inline array/js/array.js]: basically set is array in js

File: array/js/array.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+  return array.filter(function (element) {
+    var isExist = set[element];
```
Comment (NikitaSedyx) [inline array/js/array.js]: it won't work with objects
remember what type has index: String. when you do `set[element]`, toString() is called on element

File: array/js/array.js [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    return !isExist;
+  })
+}
+
+function unique2(array) {
+  var set = {};
+  var i = 0;
+  while (i < array.length) {
+    if (set[array[i]]) {
+      array.splice(i, 1);
```
Comment (NikitaSedyx) [inline array/js/array.js]: you mutate array here, it's not cool

File: array/js/array.js [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+  return array;
+}
+
+function last(array) {
+  return array[array.length - 1];
+}
+
+function excludeLast(array, count) {
+  if (count == undefined) {
+    --array.length;
```
Comment (NikitaSedyx) [inline array/js/array.js]: again, you mutates array

File: array/js/array.js [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+  } else {
+    array.length = array.length - count;
+  }
+  return array;
+}
+
+function excludeLast2(array, count) {
```
Comment (NikitaSedyx) [inline array/js/array.js]: same in this func

**Review by NikitaSedyx — CHANGES_REQUESTED** [review]

File: array/js/array.js (hunk 4 above)
Comment (vialikanau-aliaksei) [inline array/js/array.js]: compact([1, 0, 3, false, 5, NaN, null, undefined, '']) is passed to [1, 3, 5]

File: array/js/array.js [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+
+function excludeLast(array, count) {
+  if (count === undefined) {
+    count = 1;
+  }
+  return array.slice(0, array.length - count);
+
+}
+
+function excludeLast2(array, count) {
```
Comment (NikitaSedyx) [inline array/js/array.js:78]: negative count increases array length, could you make changes if count <= 0 to return new array without any changes

File: array/js/array.js [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+
+function last(array) {
+  return array[array.length - 1];
+}
+
+function excludeLast(array, count) {
+  if (count === undefined) {
+    count = 1;
+  }
+  return array.slice(0, array.length - count);
```
Comment (NikitaSedyx) [inline array/js/array.js]: same

File: array/js/array.js [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+  var array = [];
+  for (var i = start; i < end; i = i + step) {
+    array.push(i);
+  }
+  return array;
+}
+
+function compact(array) {
+  return array.filter(function (element) {
+    return element ? true : false;
```
Comment (NikitaSedyx) [inline array/js/array.js]: `return element` will work... please read why

**Review by NikitaSedyx — CHANGES_REQUESTED** [review]

**Review by NikitaSedyx — APPROVED** [review]

File: array/js/array.js [hunk 13]
```diff
@@ -0,0 +1,88 @@
+function isArray(object) {
+  return object.constructor.name == "Array";
+}
+
+function isArray2(object) {
+  return {}.toString.call(object) == "[object Array]";
```
Comment (NikitaSedyx) [inline array/js/array.js]: are you sure it works?
`SyntaxError: expected expression, got '.' `is in my console

**Review by NikitaSedyx — CHANGES_REQUESTED** [review]

File: array/js/array.js (hunk 13 above)
Comment (vialikanau-aliaksei) [reply array/js/array.js]: yep, working in edge and chrome 57. I'll to do bug fix for console

**Review by NikitaSedyx — APPROVED** [review]

**Pull request merged**